File: backend/app/schemas/academic.py
from __future__ import annotations

from pydantic import BaseModel, Field


class CourseSummary(BaseModel):
    course_id: str
    name: str
    term: str
    teacher_name: str
    description: str


class ClassSummary(BaseModel):
    class_id: str
    course_id: str
    name: str
    grade: str
    major: str
    student_count: int


class StudentSummary(BaseModel):
    student_id: str
    name: str
    student_no: str
    class_id: str
    target_path: str
    tags: list[str] = Field(default_factory=list)


class CourseListResponse(BaseModel):
    courses: list[CourseSummary]


class ClassListResponse(BaseModel):
    course_id: str
    classes: list[ClassSummary]


class StudentListResponse(BaseModel):
    class_id: str
    students: list[StudentSummary]

File: frontend/src/shared/types/academic.ts
export type CourseSummary = {
  course_id: string;
  name: string;
  term: string;
  teacher_name: string;
  description: string;
};

export type ClassSummary = {
  class_id: string;
  course_id: string;
  name: string;
  grade: string;
  major: string;
  student_count: number;
};

export type StudentSummary = {
  student_id: string;
  name: string;
  student_no: string;
  class_id: string;
  target_path: string;
  tags: string[];
};

export type CourseListResponse = {
  courses: CourseSummary[];
};

export type ClassListResponse = {
  course_id: string;
  classes: ClassSummary[];
};

export type StudentListResponse = {
  class_id: string;
  students: StudentSummary[];
};

export type AcademicImportCourse = {
  course_id: string;
  name: string;
  term?: string;
  teacher_id?: string;
  teacher_name?: string;
  teacher_no?: string;
  description?: string;
};

export type AcademicImportClass = {
  class_id: string;
  course_id: string;
  name: string;
  grade?: string;
  major?: string;
};

export type AcademicImportStudent = {
  student_id: string;
  name: string;
  student_no: string;
  class_id: string;
  course_ids?: string[];
  target_path?: string;
  tags?: string[];
};

export type AcademicImportRequest = {
  courses?: AcademicImportCourse[];
  classes?: AcademicImportClass[];
  students?: AcademicImportStudent[];
};

export type AcademicImportResponse = {
  imported_courses: number;
  imported_classes: number;
  imported_students: number;
  imported_memberships: number;
  message: string;
};
